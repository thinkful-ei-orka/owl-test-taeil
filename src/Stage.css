

.stage {
  background: #404853;
  display: flex;
  justify-content: space-around;
}
.stage-participant {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.stage-participant .name {
  background: #1F2835;
  color: #fff;
  padding: 15px;
}
.stage-participant .name h6 {
  font-size: 16px;
}
.stage-participant .name i {
  font-size: 12px;
  margin-left: 7px;
}
.stage-participant .volume {
  width: 100%;
  height: 5px;
  border-radius: 2px;
  background: #EC7779;
  margin-top: 10px;
}
.stage-participant img {
  width: 150px;
  height: 150px;
  background: #ddd;
  border-radius: 50%;
  align-self: center;
}
.stage-participant .flex-fix {
  height: 65px;
}
