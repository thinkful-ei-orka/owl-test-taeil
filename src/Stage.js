import React from 'react';

import './Stage.css';
import data from './data';

function Stage() {
  let stageParticipants = [];

  data.participants.forEach((participant) => {
    if (participant.onStage) {
      stageParticipants.push(
        <div className="stage-participant" key={participant.id}>
          <div className="name">
            {participant.name}
            <i className="fas fa-volume-up"></i>
            <i className="fas fa-search"></i>
            <div className="volume"></div>
          </div>
          <img src={participant.avatar} alt={participant.name + ' avatar'} />
          <div className="flex-fix"></div>
        </div>
      );
    }
  });

  return (
    <section className='stage'>
      {stageParticipants}
    </section>
  );
}

export default Stage;
