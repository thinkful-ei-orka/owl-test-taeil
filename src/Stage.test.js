import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Stage from './Stage'

describe(`TheDate Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    expect(
      renderer.create(<Stage />).toJSON()
    ).toMatchSnapshot()
  })
})
