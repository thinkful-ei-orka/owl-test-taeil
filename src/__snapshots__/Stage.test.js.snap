// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate Component renders the UI as expected 1`] = `
<section
  className="stage"
>
  <div
    className="stage-participant"
  >
    <div
      className="name"
    >
      Koren Templeton
      <i
        className="fas fa-volume-up"
      />
      <i
        className="fas fa-search"
      />
      <div
        className="volume"
      />
    </div>
    <img
      alt="Koren Templeton avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="flex-fix"
    />
  </div>
  <div
    className="stage-participant"
  >
    <div
      className="name"
    >
      Ashla Attwool
      <i
        className="fas fa-volume-up"
      />
      <i
        className="fas fa-search"
      />
      <div
        className="volume"
      />
    </div>
    <img
      alt="Ashla Attwool avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="flex-fix"
    />
  </div>
</section>
`;
